src/insns/acperm_32bit.adoc

```diff
@@ -29,41 +29,53 @@ ACPERM performs the following operations:
-
-The common rules are:
-
-. <<asr_perm>> cannot be set without <<x_perm>> being set
-.. Clear <<asr_perm>> unless <<x_perm>> is set
-. <<c_perm>> cannot be set without at least one of <<r_perm>> or <<w_perm>> being set.
-.. Clear <<c_perm>> unless <<r_perm>> or <<w_perm>> are set.
-. <<m_bit>> cannot be set without <<x_perm>> being set
-.. Clear <<m_bit>> unless <<x_perm>> is set
-. <<lm_perm>> cannot be set without <<c_perm>> being set
-.. Clear <<lm_perm>> unless <<c_perm>> is set.
-. <<sl_perm>> cannot be set without <<c_perm>> being set
-.. Zero <<sl_perm>> unless <<c_perm>> is set.
-. <<el_perm>> cannot be set without <<c_perm>> being set
-.. Zero <<el_perm>> unless <<c_perm>> is set.
-
-NOTE: The combination of <<x_perm>> clear and <<m_bit>> set is reserved for future extensions.
-
-The MXLEN=32 additional rules are:
-
-. Clear <<asr_perm>> unless _all_ other permissions are set
-. Clear <<c_perm>> and <<x_perm>> if <<r_perm>> is not set
-. Clear <<w_perm>> if <<c_perm>> _is_ set, but <<lm_perm>> _is not_ set.
-. Clear <<x_perm>> if <<x_perm>> and <<r_perm>> are set, and <<c_perm>> and <<w_perm>> are both clear.
-. If removing <<lm_perm>> results in unrepresentable permissions, then set <<r_perm>> and <<c_perm>> only.
+
+The rules from <<acperm_rules>> must be followed When removing permissions.
+
+[#acperm_rules]
+.ACPERM common rules
+[%autowidth,float="center",align="center",cols="2,3,3",options="header"]
+|===
+| Rule | Permission   | Only valid if
+|  1   | <<c_perm>>   | <<r_perm>> or <<w_perm>>
+3+| See <<acperm_rules_rv32>> for rules 2-5 (RV32 only)
+|  7   | <<sl_perm>>  | <<w_perm>> and <<c_perm>>
+|  8   | <<el_perm>>  | <<c_perm>> and <<r_perm>>
+|  9   | <<lm_perm>>  | <<c_perm>> and <<r_perm>>
+| 10   | <<asr_perm>> | <<x_perm>>
+| 11   | <<m_bit>>    | <<x_perm>>
+3+| See <<acperm_rules_rv32>> for rule 12 (RV32 only)
+|===
+
+The behaviour of currently illegal combinations from <<acperm_rules>> is to clear the permission if invalid (or in the case of <<sl_perm>> set it to 0 (_local_)).
+
+* For RV64 all such combinations may be redefined by future extensions.
+* For RV32 none of these combinations are possible.
+
+RV32 needs additional rules as some combinations which are legal on RV64 are not possible on RV32.
+These rules are specifically to meet the encoding requirements for <<cap_perms_encoding32>>, or
+<<cap_perms_encoding_levels32>> if <<cheri_levels_ext,{cheri_levels_ext_name}>> is implemented.
+
+[#acperm_rules_rv32]
+.ACPERM RV32 additional rules
+[%autowidth,float="center",align="center",cols="2,2,3",options="header"]
+|===
+| Rule | Permission   | Only valid if
+|  2   | <<c_perm>>   | <<r_perm>>
+|  3   | <<x_perm>>   | <<r_perm>>
+|  4   | <<w_perm>>   | not(<<c_perm>>) or (<<c_perm>> and not(<<lm_perm>>))
+|  5   | <<x_perm>>   | <<w_perm>> or <<c_perm>>
+|  6   | <<x_perm>>   | <<c_perm>> == <<lm_perm>> == <<el_perm>>
+| 12   | <<asr_perm>> | All other permissions are set.
```

Comment:
This could also be the first step, right? If we have all perms initially then the check is a no-op, and if anything is missing we also clear ASR.

Reply:
yes I think that's fair comment - if anything missing after ANDing then clear ASR